privacyidea/lib/auditmodules/sqlaudit.py

```diff
@@ -235,6 +235,10 @@ def finalize_log(self):
                             "Error occurs in action: {0!r}.".format(self.audit_data.get("action")))
                 if not "token_type" in self.audit_data:
                     self.audit_data["token_type"] = self.audit_data.get("tokentype")
+            if self.audit_data.get("startdate"):
+                duration = datetime.datetime.now() - self.audit_data.get("startdate")
```

Comment:
if there is a (isoformat)-string in "startdate", does this calculation still work?

Reply:
Yes, indeed it does!

Reply:
Correction. In the SQLAudit it actually containes a datetime object. (see my comment above).

So there is no need to change this.

Reply:
Yes, i got mixed up with the classes.

Reply:
Me too